server/runtime/src/main/java/org/infinispan/server/Server.java

```diff
@@ -536,4 +548,35 @@ public ComponentStatus getStatus() {
    public TaskManager getTaskManager() {
       return taskManager;
    }
+
+   @Override
+   public CompletionStage<Path> getServerReport() {
+      long pid = ProcessInfo.getInstance().getPid();
+      Path home = serverHome.toPath();
+      Path root = serverRoot.toPath();
+      ProcessBuilder builder = new ProcessBuilder();
+      OS os = OS.getCurrentOs();
+      switch (os) {
+         case LINUX:
+            builder.command("sh", "-c", home.resolve("bin/report.sh").toString(), Long.toString(pid), root.toString());
+            break;
+         default:
+            return CompletableFutures.completedExceptionFuture(log.serverReportUnavailable(os));
+      }
+
+      return CompletableFuture.supplyAsync(() -> {
+         try {
+            Process process = builder.start();
+            BufferedReader reader = new BufferedReader(new InputStreamReader(process.getInputStream()));
+            Path path = Paths.get(reader.readLine());
+            process.waitFor(1, TimeUnit.MINUTES);
+            return path;
+         } catch (IOException e) {
+            throw new RuntimeException(e);
+         } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+            throw new RuntimeException(e);
+         }
+      }, ForkJoinPool.commonPool());
```

Comment:
nitpick: you don't need the `ForkJoinPool.commonPool()`. It is used by default.

Reply:
Checkstyle wants it :) Blame @wburns

Reply:
Are you not able to pass the blocking executor in some way? We are trying not to use common pool at all. Tbh I should add a checkstyle to prevent calling `ForkJoinPool.commonPool` as well :)

Reply:
Should be able to just extract it from the first cacheManager.

Reply:
To be honest I am not a fan of the multiple cacheManagers in this class still. Having multiple GLOBAL components could be quite problematic.